### PR TITLE
Major refresh of CI-CD workflow and deployment setup

### PR DESCRIPTION
**Pull request by riccardoporreca**
Largely based on miraisolutions/ShinyCICD#19

This also includes monthly GithUb Actions configuration of Dependabot.

### CODE REVIEW

**Comment by riccardoporreca** [comment]
@GuidoMaggio, the app is temporarily deployed from this branch as

- https://miraisolutions.shinyapps.io/rTRhexNG-test/

which you can compare against the existing nominal deployment from the main branch

- https://miraisolutions.shinyapps.io/rTRhexNG/

The look and behavior is the same. Note that it is OK/expected for 'rsvg2' and Inkscape-related postprocessing to fail on shinyapps.io, see 066dcae

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -1,59 +1,45 @@
...
-        shell: Rscript {0}
-
-      - name: Check
-        run: rcmdcheck::rcmdcheck(args = "--no-manual", error_on = "warning", check_dir = "check")
-        shell: Rscript {0}
+      - uses: r-lib/actions/check-r-package@v2
 
       - name: Deploy to shinyapps.io
-        if: github.ref == 'refs/heads/master'
+        if: startsWith(github.ref, 'refs/heads/') || github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master'
```
Comment (riccardoporreca) [inline .github/workflows/ci-cd.yml]: Remove `startsWith(github.ref, 'refs/heads/')` before merging, to revert deploying from the main branch only

**Review by GuidoMaggio — APPROVED** [review]
ok

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (riccardoporreca) [reply .github/workflows/ci-cd.yml]: Done, as well as switching back deployment from rTRhexNG-test to rTRhexNG

**Pull request merged**